tests/test_subtemplates.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+from datetime import datetime
+import gnupg
+import mock
+import os
+import shutil
+import tempfile
+import unittest
+
+from nose.tools import assert_true
+from nose.tools import assert_equal
+from nose.tools import assert_in
+
+import settings
+from gen_cert import CertificateGen
+from test_data import NAMES
+
+
+class CertificateSubtemplateTests(unittest.TestCase):
+    """
+    Tests the subtemplates feature for certs
+
+    Subtemplates are currently written to key on designations only
+    """
+    subtemplate_courses = [
+        'Test/Subtemplates/ByDesignations',
+    ]
+
+    def test_subtemplate_courses_exist(self):
+        """
+        Check that the subtemplate courses exist in settings.CERT_DATA
+        """
+        found_sub_course_ids = [
+            course_id
+            for course_id, course_data in settings.CERT_DATA.iteritems()
+            if 'subtemplates' in course_data
+        ]
+        assert_equal(
+            set(self.subtemplate_courses),
+            set(found_sub_course_ids)
+        )
+
+    def test_subtemplate_courses_have_proper_data(self):
+        """
+        Check that subtemplates have proper key-value pairs in 'subtemplates'
+
+        Keys should be designations
+        Values should be course ids in settings.CERT_DATA
```

Comment:
I don't think we need to explain the data format here;
we're already doing so above.